visidata/loaders/_pandas.py

```diff
@@ -17,6 +17,57 @@ def open_dta(vd, p):
     if not getattr(VisiData, funcname, None):
         setattr(VisiData, funcname, lambda vd,p,ft=ft: PandasSheet(p.name, source=p, filetype=ft))
 
+@VisiData.api
+def save_dta(vd, p, *sheets):
+    import pandas as pd
+    import numpy as np
+
+    fh = open("save_dta.log", "w", encoding="utf8")
+
+    #for vs in sheets:
+    vs = sheets[0]
+
+    columns = [col.name for col in vs.visibleCols]
+    fh.write(f"columns={columns}\n")
+    fh.flush()
+
+    types = list()
+    for r_i, dispvals in enumerate(vs.iterdispvals(format=True)):
+        if r_i == 0:
+            for coli, (col, v) in enumerate(dispvals.items()):
+                if vd.isNumeric(col):
+                    types.append(float)
+                else:
+                    types.append(str)
+        continue
+
+    nrows = r_i + 1
+    ncols = len(columns)
+    fh.write(f"nrows={nrows};ncols={ncols}\n")
+    fh.flush()
+   
+    data = np.empty((nrows, len(columns)), dtype=object)
+    for r_i, dispvals in enumerate(vs.iterdispvals(format=True)):
+        for c_i, v in enumerate(dispvals.values()):
+            fh.write(f"{r_i};{c_i};{v} ({type(v)})\n")
+            fh.flush()
+            data[r_i, c_i] = v
+
+    dtype = {col:t for col,t in zip(columns, types)}
+    fh.write(f"dtype={dtype}\n")
+    fh.flush()
+
+    df = pd.DataFrame(data, columns=columns)
+    fh.write(f"Dataframe created\n")
+
+    df = df.astype(dtype)
+    fh.write(f"Data types written")
+
+    df.to_stata(p, version=118, write_index=False)
+    fh.write("DataFrame saved\n")
+
+    fh.close()
+    vd.status(f'{p} save finished')
```

Comment:
Individual savers should omit a finishing status.  I'd rather put one in `saveSheets()` so it applies to all savers.